dataframe-csv/src/test/kotlin/org/jetbrains/kotlinx/dataframe/io/DelimCsvTsvTests.kt

```diff
@@ -0,0 +1,657 @@
+        df["double"].type() shouldBe typeOf<Double?>()
+        df["number"].type() shouldBe typeOf<Double>()
+        df["time"].type() shouldBe typeOf<LocalDateTime>()
+
+        println(df)
+    }
+
+    @Test
+    fun readCsvWithFloats() {
+        val df = DataFrame.readCsv(wineCsv, delimiter = ';')
+        val schema = df.schema()
+
+        fun assertColumnType(columnName: String, kClass: KClass<*>) {
+            val col = schema.columns[columnName]
+            col.shouldNotBeNull()
+            col.type.classifier shouldBe kClass
+        }
+
+        assertColumnType("citric acid", Double::class)
+        assertColumnType("alcohol", Double::class)
+        assertColumnType("quality", Int::class)
+    }
+
+    @Test
+    fun `read standard CSV with floats when user has alternative locale`() {
+        val currentLocale = Locale.getDefault()
+        try {
+            Locale.setDefault(Locale.forLanguageTag("ru-RU"))
+            val df = DataFrame.readCsv(wineCsv, delimiter = ';')
+            val schema = df.schema()
+
+            fun assertColumnType(columnName: String, kClass: KClass<*>) {
+                val col = schema.columns[columnName]
+                col.shouldNotBeNull()
+                col.type.classifier shouldBe kClass
+            }
+
+            assertColumnType("citric acid", Double::class)
+            assertColumnType("alcohol", Double::class)
+            assertColumnType("quality", Int::class)
+        } finally {
+            Locale.setDefault(currentLocale)
+        }
+    }
+
+    @Test
+    fun `read with custom header`() {
+        val header = ('A'..'K').map { it.toString() }
+        val df = DataFrame.readCsv(simpleCsv, header = header, skipLines = 1)
+        df.columnNames() shouldBe header
+        df["B"].type() shouldBe typeOf<Int>()
+
+        val headerShort = ('A'..'E').map { it.toString() }
+        val dfShort = DataFrame.readCsv(simpleCsv, header = headerShort, skipLines = 1)
+        dfShort.columnsCount() shouldBe 5
+        dfShort.columnNames() shouldBe headerShort
+    }
+
+    @Test
+    fun `read first rows`() {
+        val expected =
+            listOf(
+                "untitled",
+                "user_id",
+                "name",
+                "duplicate",
+                "username",
+                "duplicate1",
+                "duplicate11",
+                "double",
+                "number",
+                "time",
+                "empty",
+            )
+        val dfHeader = DataFrame.readCsv(simpleCsv, readLines = 0)
+        dfHeader.rowsCount() shouldBe 0
+        dfHeader.columnNames() shouldBe expected
+
+        val dfThree = DataFrame.readCsv(simpleCsv, readLines = 3)
+        dfThree.rowsCount() shouldBe 3
+
+        val dfFull = DataFrame.readCsv(simpleCsv, readLines = 10)
+        dfFull.rowsCount() shouldBe 5
+    }
+
+    @Test
+    fun `if string starts with a number, it should be parsed as a string anyway`() {
```

Comment:
I wanted to see actual file to understand what name implies. Maybe it'd be better to inline the file and use readCsvStr